primap2/_aggregate.py

```diff
@@ -267,51 +294,109 @@ def aggregate_coordinates(
+                        rule_tolerance = tolerance
+                    if "filter" in rule.keys():
+                        filter = rule.pop("filter")
+                        if "variable" in filter.keys():
+                            if da_out.name in filter["variable"]:
+                                filter.pop("variable")
+                            else:
+                                continue
+                        if "entity" in filter.keys():
+                            if da_out.attrs["entity"] in filter["entity"]:
+                                filter.pop("entity")
+                            else:
+                                continue
+                    else:
+                        filter = {}
+                elif isinstance(rule, list):
+                    source_values = rule
+                    filter = {}
                     rule_tolerance = tolerance
-                if "filter" in rule.keys():
-                    filter = rule.pop("filter")
                 else:
-                    filter = {}
-
-                source_values = rule.pop("sources")
+                    logger.error(
+                        "Unrecognized aggregation definition for "
+                        f"{value_to_aggregate!r}"
+                    )
+                    raise ValueError(
+                        "Unrecognized aggregation definition for "
+                        f"{value_to_aggregate!r}"
+                    )
 
-                filter.update({coordinate: source_values})
-                data_agg = da_out.pr.loc[filter].pr.sum(
-                    dim=coordinate, skipna=True, min_count=1
-                )
-                if not data_agg.isnull().all().data:
-                    data_agg = data_agg.expand_dims([full_coord_name])
-                    data_agg = data_agg.assign_coords(
-                        coords={
-                            full_coord_name: (full_coord_name, [value_to_aggregate])
-                        }
+                # check if all source values present
+                values_present = list(da_out[full_coord_name].values)
+                source_values_present = [
+                    val for val in source_values if val in values_present
+                ]
+                missing_values = list(set(source_values) - set(source_values_present))
+                if len(missing_values) > 0:
+                    logger.info(
+                        f"Not all source values present for "
+                        f"{value_to_aggregate!r} in coordinate {full_coord_name!r}. "
+                        f"Missing: {missing_values}"
+                    )
+                if source_values_present:
+                    filter.update({coordinate: source_values_present})
+                    data_agg = da_out.pr.loc[filter].pr.sum(
+                        dim=coordinate, skipna=skipna, min_count=min_count
+                    )
+                    if not data_agg.isnull().all().data:
```

Comment:
What does `.data` do in this context? It is used to access the underlying datastructure for xarray, which in our context usually is a numpy array, but could also be something else, like a dask array or a netcdf file backed array or whatever. Maybe you don't need it at all, or just need to call `item()` to get a simple bool?

Reply:
It works without `.data` Maybe it was leftover from code restructuring